server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueComputation.java

```diff
@@ -51,16 +58,23 @@ public void afterReportProcessing() {
   }
 
   private void guessAuthor(DefaultIssue issue) {
-    // TODO
+    if (issue.line() != null) {
+      issue.setAuthorLogin(linesCache.lineAuthor(issue.line()));
+    }
   }
 
   private void autoAssign(DefaultIssue issue) {
-    // TODO
+    String scmAccount = issue.authorLogin();
+    if (scmAccount != null) {
+      issue.setAssignee(scmAccountCache.getNullable(scmAccount));
+    }
   }
 
   private void copyRuleTags(DefaultIssue issue) {
-    RuleDto rule = ruleCache.get(issue.ruleKey());
-    issue.setTags(Sets.union(rule.getTags(), rule.getSystemTags()));
+    RuleDto rule = ruleCache.getNullable(issue.ruleKey());
```

Comment:
it was null at a time in tests, but it's not the case anymore. And not a good reason. Fixed.